### ISSUE TITLE
Learning rate schedules do not work with JAX backend

### ISSUE THREAD

**Issue by martin-gorner** [keras-team/keras-core#213]
repro colab: https://colab.corp.google.com/drive/15AFfhJIG9p3kgxjZQBxpns7wmwsOMzDi?resourcekey=0-ZgBZ6c0U2RjfstKTmOrO-g&usp=sharing

The Colab displays the learning rate from the optimizer through a custom metric.

Repro steps:
* select JAX backend in first cell, run notebook. Reported LR is:

Epoch 1/5 val_lr_metric: 0.0100
Epoch 2/5 val_lr_metric: 0.0100
Epoch 3/5 val_lr_metric: 0.0100

* restart from scratch (use "disconnect and delete runtime"), then  select tensorflow backend in the first cell. Reported LR is:

Epoch 1/5 val_lr_metric: 0.0060
Epoch 2/5 val_lr_metric: 0.0036
Epoch 3/5 val_lr_metric: 0.0022
